app/code/Magento/Inventory/Model/StockIndexManager.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Inventory\Model;
+
+use Magento\Framework\MultiDimensionalIndex\IndexName;
+use Magento\Framework\MultiDimensionalIndex\IndexNameBuilder;
+use Magento\Inventory\Indexer\Stock\StockIndexer;
+
+/**
+ * Manager for stock index.
+ */
+class StockIndexManager
+{
+    /**
+     * @var string
+     */
+    private $dimensionName = 'stock_';
+
+    /**
+     * @var IndexNameBuilder
+     */
+    private $indexNameBuilder;
+
+    /**
+     * @param IndexNameBuilder $indexNameBuilder
+     */
+    public function __construct(
+        IndexNameBuilder $indexNameBuilder
+    ) {
+        $this->indexNameBuilder = $indexNameBuilder;
+    }
+
+    /**
+     * Build index by stock id and alias.
+     *
+     * @param string $stockId
+     * @param string $alias
+     *
+     * @return IndexName
+     */
+    public function buildIndex(string $stockId, string $alias): IndexName
```

Comment:
`alias` is case related to index part 
So it is shouldn't be part of API

Reply:
Done.